esmvalcore/preprocessor/_io.py

```diff
@@ -180,10 +193,13 @@ def load(file, callback=None, ignore_warnings=None):
         with suppress_errors():
             raw_cubes = iris.load_raw(file, callback=callback)
     logger.debug("Done with loading %s", file)
+
     if not raw_cubes:
         raise ValueError(f'Can not load cubes from {file}')
+
     for cube in raw_cubes:
-        cube.attributes['source_file'] = file
+        cube.attributes['source_file'] = str(file)
+
```

Comment:
I would put a check on `file`'s type in here (str is a bit too general, so we are exposing us to fails if load is called with `load("s3://cow")`) - a placeholder so we maybe get support for eg object storage "files" in the future

Reply:
Sorry, I don't fully get your comment - at the moment, only valid `Path` and `str` objects can be passed to `load`. Since a couple of version iris can digest `Path` objects, that's why I am calling `Path(file)` at the beginning of the function now. Here, we needs `str(file)` for the attribute to be properly set.

Since now other types are supported, I don't think we need to add additional checks here. If the path is invalid, we will either get an iris error during `iris.load_raw` or a `ValueError` from our code.

Reply:
indeed, and my second phrasing of the comment was as vague as a Tory declaring their taxes - what I really meant was we should probably think of examining the `str` path a bit more in the context when it is not a valid string of a POSIX path, and throw up the error, but a little bit more tailored towards cases such eg the str path starts with "s3" or "https" - the first steps towards supporting remote/object store files. Clearly, not a priority now, so totally up to you if you want to go start going down this rabbit hole :rabbit: